txscript/data/script_tests.json

```diff
@@ -1365,87 +1459,193 @@
-["0", "IF 0x50 ENDIF 1", "NONE", "OK", "0x50 is reserved (ok if not executed)"],
-["0x51", "0x5f ADD 0x60 EQUAL", "NONE", "OK", "0x51 through 0x60 push 1 through 16 onto stack"],
-["1","NOP", "NONE", "OK"],
-["0", "IF VER ELSE 1 ENDIF", "NONE", "OK", "VER non-functional (ok if not executed)"],
-["0", "IF RESERVED ELSE 1 ENDIF", "NONE", "OK", "RESERVED ok in un-executed IF"],
-
-
-
-["2 -2 ADD", "0 EQUAL", "NONE", "OK"],
-["2147483647 -2147483647 ADD", "0 EQUAL", "NONE", "OK"],
-["-1 -1 ADD", "-2 EQUAL", "NONE", "OK"],
-
-["0 0","EQUAL", "NONE", "OK"],
-["1 1 ADD", "2 EQUAL", "NONE", "OK"],
-["1 1ADD", "2 EQUAL", "NONE", "OK"],
-["111 1SUB", "110 EQUAL", "NONE", "OK"],
-["111 1 ADD 12 SUB", "100 EQUAL", "NONE", "OK"],
-["0 ABS", "0 EQUAL", "NONE", "OK"],
-["16 ABS", "16 EQUAL", "NONE", "OK"],
-["-16 ABS", "-16 NEGATE EQUAL", "NONE", "OK"],
-["0 NOT", "NOP", "NONE", "OK"],
-["1 NOT", "0 EQUAL", "NONE", "OK"],
-["11 NOT", "0 EQUAL", "NONE", "OK"],
-["0 0NOTEQUAL", "0 EQUAL", "NONE", "OK"],
-["1 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
-["111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
-["-111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
-["1 1 BOOLAND", "NOP", "NONE", "OK"],
+["BOOLAND test coverage"],
+["NOP BOOLAND", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLAND requires two items on the stack and NOP must not be treated as one"],
+["BOOLAND", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLAND requires two items on the stack, stack has no items"],
+["0 BOOLAND", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLAND requires two items on the stack, stack has one item"],
+["1 1 BOOLAND", "NOP", "NONE", "OK", "BOOLAND must perform a logical and operation on the boolean values of the first two stack items"],
 ["1 0 BOOLAND", "NOT", "NONE", "OK"],
 ["0 1 BOOLAND", "NOT", "NONE", "OK"],
 ["0 0 BOOLAND", "NOT", "NONE", "OK"],
-["16 17 BOOLAND", "NOP", "NONE", "OK"],
-["1 1 BOOLOR", "NOP", "NONE", "OK"],
+["1 2147483648 BOOLAND", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
+
+["BOOLOR test coverage"],
+["NOP BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two input values and NOP must not be treated as one"],
+["BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two input values, stack has no items"],
+["0 BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two items on the stack, stack has no items"],
+["1 1 BOOLOR", "NOP", "NONE", "OK", "BOOLOR must perform a logical and operation the boolean values of the first two stack items"],
 ["1 0 BOOLOR", "NOP", "NONE", "OK"],
 ["0 1 BOOLOR", "NOP", "NONE", "OK"],
 ["0 0 BOOLOR", "NOT", "NONE", "OK"],
-["16 17 BOOLOR", "NOP", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUAL", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK"],
-["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK"],
+["1 2147483648 BOOLOR", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
```

Comment:
Same here regarding needing another test for the other operand being >4 bytes and comment.